db/sqlinterfaces.c

```diff
@@ -5361,12 +5361,11 @@ static int enqueue_sql_query(struct sqlclntstate *clnt)
     }
 
     q_depth_tag_and_sql = thd_queue_depth();
-    if (thdpool_get_nthds(pool) == thdpool_get_maxthds(pool))
+    /* If all threads are busy, the request itself will likely be queued. Count it in. */
+    if (thdpool_get_nbusythds(pool) == thdpool_get_maxthds(pool))
```

Comment:
We misreport queue-depth, when sqlenginepool's number of threads (which could all be free) equals its `maxt`. This can happen if a database has a burst which uses `maxt` many threads, and has very little load afterwards.

Reply:
See [this link](https://grafana.prod.bloomberg.com/d/b3682oWiz/comdb2-stats?viewPanel=32&orgId=1&from=now-24h&to=now&var-database=ma2secdb&var-tier=prod&var-cluster_txt=c2mgpa&var-host=All&var-stage_txt=All&var-Shift=0w) for an example of miscounting - the database constantly has a queue depth of 1.

Reply:
Thanks Rivers, I think this is the biggest problem in how we report queue depth/time.